src/androidTest/java/de/blau/android/layer/LayerDialogTest.java

```diff
@@ -262,4 +264,35 @@ public void backgroundLayer() {
         assertNotNull(layer);
         assertEquals(TileLayerSource.LAYER_MAPNIK, layer.getTileLayerConfiguration().getId());
     }
+
+    /**
+     * Checks if filters are correct
+     */
+    @Test
+    public void layerFilter() {
+        assertTrue(TestUtils.clickResource(device, true, device.getCurrentPackageName() + ":id/layers", true));
+        assertTrue(TestUtils.clickButton(device, device.getCurrentPackageName() + ":id/add", true));
+        assertTrue(TestUtils.clickText(device, false, main.getString(R.string.menu_layers_add_backgroundlayer), true, false));
+
+        ViewBox viewBox = map.getViewBox();
+        String[] aerialIds = TileLayerSource.getIds(viewBox, true, TileLayerSource.Category.photo);
+        String[] terrainIds = TileLayerSource.getIds(viewBox, true, TileLayerSource.Category.elevation);
+
+        String[] aerialNames = TileLayerSource.getNames(aerialIds);
+        String[] terrainNames = TileLayerSource.getNames(terrainIds);
+
+        assertTrue(TestUtils.clickText(device, true, "All", true, false));
+
+        assertTrue(TestUtils.clickText(device, true, "Aerial imagery", true, false));
+        UiObject2 bingAerial = TestUtils.findObjectWithText(device, false, "Bing aerial imagery", 1000);
+        List<UiObject2> childrenAerial = bingAerial.getParent().getChildren();
+        assertTrue(TestUtils.findInParentList(aerialNames, childrenAerial.get(0).getText()));
+
+        assertTrue(TestUtils.clickText(device, true, "Terrain", true, false));
+        UiObject2 stamenTerrain = TestUtils.findObjectWithText(device, false, "Stamen Terrain", 1000);
```

Comment:
This is a bit brittle as there is no guarantee that the Stamen Terrain layer is available (in general you can only rely on the couple of entries that are Vespucci specific to be around). Best simply add a layer of the Terrain category with TileLayerSource.addOrUpdateCustomLayer at the beginning of the test with an unique name and check if that then turns up.